### ISSUE TITLE
failing tests on master

### ISSUE THREAD

**Issue by ThomasLecocq** [obspy/obspy#2977]
I suspect the current error on master is linked to:
> More than 20 figures have been opened. Figures created through the pyplot interface (`matplotlib.pyplot.figure`) are retained until
> explicitly closed and may consume too much memory. (To control this warning, see the rcParam `figure.max_open_warning`).

which appears 2 times (making the warning count to 4, breaking a 2==4 assertion)

should we add some `plt.close("all")` at some locations in the test suite?

**Comment by trichter** [comment]
May be we can add a test function decorator that automatically closes figures

```py
def close_fig(test):
    @functools.wraps(test)
    def new_test(*args, **kwargs):
        try:
            return test(*args, **kwargs)
        finally:
            plt.close()  # or plt.close('all')
    return new_test
```

**Comment by trichter** [comment]
I'll create a PR.

**Comment by trichter** [comment]
Fixed in #2978
